### PR TITLE
Update and cleanup documentation

### PR DESCRIPTION
**Pull request by pramodk**
Update and cleanup documentaiton
 * remove unnecessary info from README
 * remove .gitreview from gerrit
 * CREDIT.txt is no longer required
 * added .gitignore
 * update mod2c to master

CI_BRANCHES:NEURON_BRANCH=master,

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+.DS_Store
+*.swp
+.idea
+*.srctrl*
+
+# HPC coding conventions
+.clang-format
+.clang-tidy
+.cmake-format.yaml
+.pre-commit-config.yaml
```
Comment (nrnhines) [inline .gitignore:14]: Did you intend to add this file?  Though most seem generically useful to ignore, one seems somewhat personal, e.g. ```.idea``` .  Some seem like one would like to ```git clean -d -f``` away.

**Review by nrnhines — APPROVED** [review]

File: .gitignore (hunk 1 above)
Comment (pramodk) [reply .gitignore:14]: `.idea` is folder created by editors like IntelliJ and hence it's ignored. To remove everything including ignored files/folder, one can just add `-x` flag to `git clean`

**Comment by pramodk** [comment]
Retest this please

**Comment by pramodk** [comment]
Plasticity test is failing after mod2c update? need to check if/how that's causing the spike changes!

**Comment by pramodk** [comment]
retest this please

File: .gitignore (hunk 1 above)
Comment (alkino) [reply .gitignore:14]: For things personal included by your IDE, you should modify the .gitignore in your home

**Review by alkino — APPROVED** [review]

File: .gitignore (hunk 1 above)
Comment (pramodk) [reply .gitignore:13]: fixed

**Comment by pramodk** [comment]
Retest this please

**Comment by pramodk** [comment]
retest this please

**Comment by pramodk** [comment]
retest this please

**Pull request merged**